File: 백준/Gold/2170. 선 긋기/선 긋기.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
    cin.tie(0)->sync_with_stdio(0);

    int n;
    cin>>n;
    vector<pair<int,int>>arr(n);
    for (int i=0; i<n; i++) {
        cin>>arr[i].first>>arr[i].second;
    }
    sort(arr.begin(),arr.end());

    int ans=0;
    int l=INT_MIN,r=INT_MIN;
    for (int i=0; i<n; i++) {
        if (r<arr[i].first) {
            ans+=r-l;
            l=arr[i].first;
        }
        if (r<arr[i].second) {
            r=arr[i].second;
        }
    }
    ans+=r-l;
    cout<<ans;
}

File: 백준/Gold/2170. 선 긋기/선 긋기.py
import sys

input=sys.stdin.readline
n = int(input())

arr=[]

for _ in range(n):
    x,y=map(int,input().split())
    arr.append((x,y))

arr.sort()

ans=0
l=-10**18
r=-10**18

for x,y in arr:
    if r<x:
        ans+=r-l
        l=x
    if r<y:
        r=y

ans+=r-l
print(ans)